src/components/TableDialog.vue

```diff
@@ -218,10 +218,7 @@ export default defineComponent({
       }));
     });
 
-    const edgeTable = computed(() => {
-      const keys = Object.keys(sampleRows.value[0] || {});
-      return keys.includes('_from') && keys.includes('_to');
-    });
+    const edgeTable = computed(() => Object.prototype.hasOwnProperty.call(sampleRows.value[0], '_from') && Object.prototype.hasOwnProperty.call(sampleRows.value[0], '_to'));
```

Comment:
For readability, you may want to split this up into a couple of `const` expressions, like:

```typescript
const hasFrom = Object.prototype.hasOwnProperty.call(sampleRows.value[0]);
```

In general, when you have complex expressions composed into a long line of code, it's good to consider the structure of it for purposes of readability to other devs (including Emi-from-the-future).

I think you also want to extract the `keys` value as in your old code:

```typescript
const sample = sampleRows.value[0] || {};
```

so that you aren't operating on `undefined` by accident. Essentially, you want the type of `sample` to be `Node` rather than `Node | undefined` (I'm saying `Node` because I forgot the true name of the type, but my point stands).

Reply:
Thank you for the feedback, those are both great points!

See a970386